packages/core/src/LernaTool.ts

```diff
@@ -0,0 +1,92 @@
+import path from 'path';
+import fs from 'fs-extra';
+
+import { Tool, ToolType, Package, PackageJSON, Packages, InvalidMonorepoError } from './Tool';
+import { expandPackageGlobs, expandPackageGlobsSync } from "./expandPackageGlobs";
+
+export interface LernaJson {
+    useWorkspaces?: boolean;
+    packages?: string[];
+}
+
+export const LernaTool: Tool = {
+    type: 'lerna',
+
+    async isMonorepoRoot(directory: string): Promise<boolean> {
+        try {
+            const lernaJson = await fs.readJson(path.join(directory, "lerna.json")) as LernaJson;
+            if (lernaJson.useWorkspaces !== true) {
+                return true;
+            }
+        } catch (err: any) {
+            if (err.code !== "ENOENT") {
+                throw err;
+            }
+        }
+        return false;
+    },
+
+    isMonorepoRootSync(directory: string): boolean {
+        try {
+            const lernaJson = fs.readJsonSync(path.join(directory, "lerna.json")) as LernaJson;
+            if (lernaJson.useWorkspaces !== true) {
+                return true;
+            }
+        } catch (err: any) {
+            if (err.code !== "ENOENT") {
+                throw err;
+            }
+        }
+        return false;
+    },
+
+    async getPackages(directory: string): Promise<Packages> {
```

Comment:
an interesting fact about this one - this can be successfully called even when `isMonorepoRoot` returns `false` (since `getPackages` doesn't validate again the `useWorkspaces` flag). We won't be doing that since we'll always check `isMonorepoRoot` beforehand but I wonder if this is OK in terms of the design.